frappe/website/website_generator.py

```diff
@@ -129,4 +139,34 @@ def send_indexing_request(self, operation_type='URL_UPDATED'):
 
 			url = frappe.utils.get_url(self.route)
 			frappe.enqueue('frappe.website.doctype.website_settings.google_indexing.publish_site', \
-				url=url, operation_type=operation_type)
\ No newline at end of file
+				url=url, operation_type=operation_type)
+
+	# Override this method to disable indexing
+	def allow_website_search_indexing(self):
+		return bool(self.route)
```

Comment:
Check for `is_published_field`, and check if it is truthy.
Add a field in Has Web View section of doctype `Index Web Pages for Search`

Reply:
The `allow_website_search_indexing` check is also referred when removing unpublished documents from the index. Checking for `is_published_field` will break this.

> Add a field in Has Web View section of doctype `Index Web Pages for Search`

Yep, this is cool

Reply:
@netchampfaris please check if this needs more thought

Reply:
@scmmishra I mean `allow_website_search_indexing` should check if there is a route and if the page is published. This will handle a lot of use-cases.

Reply:
`allow_website_search_indexing`

This checks if at all search index operations can be performed on this doc or not…. One of the operations is to remove the index if unpublished